lib/mix/lib/mix/task.compiler.ex

```diff
@@ -312,4 +312,26 @@ defmodule Mix.Task.Compiler do
 
     Mix.Sync.PubSub.broadcast(build_path, lazy_message)
   end
+
+  @doc """
+  Reenables given compilers so they can be executed again down the stack.
+
+  If an umbrella project reenables compilers, they are re-enabled for all
+  child projects.
+
+  Default is `:all` which effectively means all the compilers returned by
+  `Mix.Task.Compiler.compilers()` are to be reenabled. This task always
+  re-enables `"compile"` and `"compile.all"`.
+  """
+  @spec reenable([{:compilers, compilers}]) :: :ok when compilers: :all | [Mix.Task.task_name()]
```

Comment:
I don't think we should use task_name, right? Because it is the compiler name, not the task one.

```suggestion
  @spec reenable([compilers: :all]) :: :ok when compilers: :all | [atom()]
```